### PR TITLE
Add utility class which contains methods which helps to convert HTTP1…

### PR DESCRIPTION
**Pull request by normanmaurer**
…/x to BHTTP messages.

Motivation:

Often people need to translate from HTTP1/x to BHTTP, we should provide utilities for this.

Modifications:

Add BinaryHttpConvert class which contain various static helper methods

Result:

Easily be able to convert from HTTP1/x to BHTTP.

### CODE REVIEW

**Review by gaosa — APPROVED** [review]

**Review by thomdev — COMMENTED** [review]
LGTM

File: codec-bhttp/src/main/java/io/netty/incubator/codec/bhttp/BinaryHttpConverter.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+     * @return          the created request.
+     */
+    public static FullBinaryHttpRequest convert(FullHttpRequest request, String scheme, String authority) {
+        BinaryHttpHeaders headers = copyAndSanitize(request.headers());
+        BinaryHttpHeaders trailers = copyAndSanitize(request.trailingHeaders());
+
+        FullBinaryHttpRequest binaryHttpRequest =  new DefaultFullBinaryHttpRequest(request.protocolVersion(),
+                request.method(), scheme, authority, request.uri(), request.content().retain(), headers, trailers);
+        binaryHttpRequest.setDecoderResult(request.decoderResult());
+        request.release();
```
[the remark this answers is not in the document]
Comment (normanmaurer) [reply codec-bhttp/src/main/java/io/netty/incubator/codec/bhttp/BinaryHttpConverter.java:78]: Good idea

[the remark this answers is not in the document]
Comment (normanmaurer) [reply codec-bhttp/src/main/java/io/netty/incubator/codec/bhttp/BinaryHttpConverter.java:78]: https://github.com/netty/netty-incubator-codec-ohttp/pull/83

File: codec-bhttp/src/main/java/io/netty/incubator/codec/bhttp/BinaryHttpConverter.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        }
+        BinaryHttpHeaders headers = copyAndSanitize(response.headers());
+        BinaryHttpResponse binaryHttpResponse = new DefaultBinaryHttpResponse(
+                response.protocolVersion(), response.status(), headers);
+        binaryHttpResponse.setDecoderResult(response.decoderResult());
+        return binaryHttpResponse;
+    }
+
+    /**
+     * Creates a {@link FullBinaryHttpResponse} from the given {@link FullHttpRequest}.
```
Comment (gaosa) [inline codec-bhttp/src/main/java/io/netty/incubator/codec/bhttp/BinaryHttpConverter.java:103]: `{@link FullHttpRequest}` -> `{@link FullHttpResponse}`

**Pull request merged**